neqo-transport/src/connection.rs

```diff
@@ -1561,26 +1676,106 @@ impl Connection {
                 }
             }
         } else {
-            debug_assert_eq!(self.role, Role::Server);
             Ok(())
         }
     }
 
-    /// Process the final set of transport parameters.
-    fn process_tps(&mut self) -> Res<()> {
-        self.validate_odcid()?;
-        if let Some(token) = self
-            .tps
-            .borrow()
-            .remote
-            .as_ref()
-            .unwrap()
-            .get_bytes(tparams::STATELESS_RESET_TOKEN)
-        {
-            let reset_token = <[u8; 16]>::try_from(token).unwrap().to_owned();
-            self.path.as_mut().unwrap().set_reset_token(reset_token);
+    fn validate_cids_draft_28_plus(&mut self) -> Res<()> {
+        let tph = self.tps.borrow();
```

Comment:
You call remote() on this three times, so maybe this can instead be:

```suggestion
        let remote_tps = self.tps.borrow().remote.as_ref().unwrap();
```

Note that you need to not call `remote()` as that picks up 0-RTT transport parameters.

Reply:
nice!